packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -372,4 +373,17 @@ export class BaseInvocationScope<TReturn = any> {
 
     return provider;
   }
+
+  /**
+   * Obtains the ID of a transaction.
+   *
+   * @param chainId - the chainId to use to hash the transaction with
+   * @returns the ID of the transaction.
+   */
+  async getTransactionId(chainId?: number): Promise<string> {
+    const chainIdToHash = chainId ?? (await this.getProvider().getChainId());
+
+    const transactionRequest = await this.getTransactionRequest();
+    return hashTransaction(transactionRequest, chainIdToHash);
+  }
```

Comment:
Maybe it would be nice to have this implemented on `BaseTransactionRequest`, this way it would also be possible to get the `TxID` when using custom transactions created with `new ScriptTransactionRequest` or `new CreateTransactionRequest`. 

Then in `BaseInvocationScope` we could simply call `transactionRequest.getTransactionId(chainId)`.

Just an idea, if you feel that does not make sense, we can skip that.

Reply:
This is a good idea, going to reference #1465 as I would like to consolidation tx preparation there, and @Dhaiwat10 is also working #1451 which I believe brings a txId function to `ScriptTransactionRequest`.